scripts/globals/combat/distance_correction.lua

```diff
@@ -0,0 +1,182 @@
+-- https://ffxiclopedia.fandom.com/wiki/Distance#Distance_and_Ranged_Attack
+xi.combat.distanceCorrection.table =
+{
+    LONGBOW =
+    {
+        { range =  1.00,
+        func = function(dist)
+            return 65  -- <=1'
+        end
+
+        },
+        { range =  7.49,
+        func = function(dist)
+            return 0.10641 * dist^3 - 0.582692 * dist^2 + 3.44744 * dist + 62.0288 -- <7.5'
+        end
+
+        },
+        { range = 10.50,
+        func = function(dist)
+            return 100 -- Sweet Spot from 7.5' to 10.5'
+        end
+
+        },
+        { range = 25.00,
+        func = function(dist)
+            return -0.0071708 * dist^3 + 0.450248 * dist^2 - 9.72569 * dist + 160.781 -- <=15.0'
+        end
+
+        },
+        { range = 100.0,
+        func = function(dist)
+            return 87  -- Default to >25'
+        end
+
+        },
+    },
+
+    CROSSBOW =
+    {
+        { range =  1.00,
+        func = function(dist)
+            return 65  -- <=1'
+        end
+
+        },
+        { range =  5.99,
+        func = function(dist)
+            return 0.846734 * dist^2 + 0.796482 * dist + 63.7337 -- <6'
+        end
+
+        },
+        { range = 10.00,
+        func = function(dist)
+            return 100  -- Sweet Spot from 6' to 10'
+        end
+
+        },
+        { range = 20.00,
+        func = function(dist)
+            return 0.12 * dist^2 - 5 * dist + 138 -- <=20'
+        end
+
+        },
+        { range = 100.0,
+        func = function(dist)
+            return 86 -- Default for >20'
+        end
+
+        },
+    },
+
+    GUN =
+    {
+        { range =  1.00,
+        func = function(dist)
+            return 75 -- <=1'
+        end
+
+        },
+        { range =  4.49,
+        func = function(dist)
+            return 1.42857 * dist^2 - 0.714286 * dist + 74.2857  -- <4.5'
+        end
+
+        },
+        { range =  6.00,
+        func = function(dist) -- Sweet spot from 4.5' to 6.0'
+            return 100
+        end
+
+        },
+        { range = 15.00,
+        func = function(dist)
+            return -0.0133333 * dist^3 + 0.646667 * dist^2 - 10.2333 * dist + 141 -- <=15'
+        end
+
+        },
+        { range = 25.00,
+        func = function(dist)
+            return dist^2 / 50 - (11 * dist) / 10 + 100  -- <=20'
+        end
+
+        },
+        { range = 100.0,
+        func = function(dist)
+            return 85  -- Default for >20'
+        end
+
+        },
+    },
+}
+
+xi.combat.distanceCorrection.applyCurve = function(distance, curve)
+    for i = 1, #curve do
+        if distance <= curve[i].range then
+            return utils.clamp(curve[i].func(distance), 65, 100) / 100
+        end
+    end
+
+    -- Fallback
+    return 1.00
+end
+
+xi.combat.distanceCorrection.throwing = function(distance)
+    if distance <= 1.00 then
+        return 1.00 -- Sweet Spot Under or at 1'
+    end
+
+    return 1.00 - ((1.00 * distance) / 100) -- Lose 1%/yalm
+end
```

Comment:
`return 1.00 - ((1.00 * distance) / 100)`
vs
`return 1 - distance / 100`

Any difference between both?
Also, in general, any reason to have floats all around in lua?

Reply:
> `return 1.00 - ((1.00 * distance) / 100)` vs `return 1 - distance / 100`
> 
> Any difference between both? Also, in general, any reason to have floats all around in lua?

Ah, I think that was translated from c++ code originally so probably had leftover float logic from there. I cleaned up the file now so fewer floats. Thanks!